tools/gulp-tasks/cldr/extract.js

```diff
@@ -72,62 +78,104 @@ module.exports = (gulp, done) => {
+    console.log(`\t${I18N_GLOBAL_FOLDER}/${locale}.ts`);
+    fs.writeFileSync(
+        `${RELATIVE_I18N_GLOBAL_FOLDER}/${locale}.js`,
+        generateGlobalLocale(
+            locale, locale === 'en' ? new cldrJs('en') : localeData, baseCurrencies));
+
   });
   console.log(`${LOCALES.length} locale files generated.`);
 
   console.log(`All i18n cldr files have been generated, formatting files..."`);
   const format = require('gulp-clang-format');
   const clangFormat = require('clang-format');
   return gulp
-    .src([
-      `${I18N_DATA_FOLDER}/**/*.ts`,
-      `${I18N_FOLDER}/currencies.ts`,
-      `${I18N_CORE_FOLDER}/locale_en.ts`
-    ], {base: '.'})
-    .pipe(format.format('file', clangFormat))
-    .pipe(gulp.dest('.'));
+      .src(
+          [
+            `${I18N_DATA_FOLDER}/**/*.ts`, `${I18N_FOLDER}/currencies.ts`,
+            `${I18N_CORE_FOLDER}/locale_en.ts`
+          ],
+          {base: '.'})
+      .pipe(format.format('file', clangFormat))
+      .pipe(gulp.dest('.'));
 };
 
 /**
- * Generate file that contains basic locale data
+ * Generate file that contains all locale data stored in a global
  */
 function generateLocale(locale, localeData, baseCurrencies) {
-  // [ localeId, dateTime, number, currency, pluralCase ]
-  let data = stringify([
-    locale,
-    ...getDateTimeTranslations(localeData),
-    ...getDateTimeSettings(localeData),
-    ...getNumberSettings(localeData),
-    ...getCurrencySettings(locale, localeData),
-    generateLocaleCurrencies(localeData, baseCurrencies)
-  ], true)
-  // We remove "undefined" added by spreading arrays when there is no value
-    .replace(/undefined/g, 'u');
-
-  // adding plural function after, because we don't want it as a string
-  data = data.substring(0, data.lastIndexOf(']')) + `, plural]`;
-
   return `${HEADER}
 const u = undefined;
 
 ${getPluralFunction(locale)}
 
-export default ${data};
+export default ${generateBasicLocaleString(locale, localeData, baseCurrencies)};
 `;
 }
 
 /**
- * Generate a file that contains extra data (for now: day period rules, and extended day period data)
+ * Generate the extra data
  */
 function generateLocaleExtra(locale, localeData) {
+  return `${HEADER}
+const u = undefined;
+
+export default ${generateDayPeriodsSupplementalString(locale, localeData)};
+`;
+}
+
+function generateGlobalLocale(locale, localeData, baseCurrencies) {
+  const basicLocaleData = generateBasicLocaleString(locale, localeData, baseCurrencies);
+  const extraLocaleData = generateDayPeriodsSupplementalString(locale, localeData);
+  const data = basicLocaleData.replace(/\]$/, `, ${extraLocaleData}]`);
+  return `${HEADER}
+(function() {
+  const root = typeof globalThis !== 'undefined' && globalThis || typeof global !== 'undefined' && global || typeof window !== 'undefined' && window;
+  root.ng = root.ng || {};
+  root.ng.common = root.ng.common || {};
+  root.ng.common.locale = root.ng.common.locale || {};
```

Comment:
the namespace should be "locales" not "locale". Can you please fix?
```suggestion
  root.ng.common.locale = root.ng.common.locales || {};
```

Reply:
this is because the ESM entry point is `@angular/common/locales`

Reply:
👍